Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM node:buster AS builder
+
+RUN apt-get update
+RUN apt-get install -y git
+
+RUN git clone https://github.com/codetheweb/MTU-Transfer-Course-Gatherer.git /app/
```

Comment:
Should we be cloning here, or can we just do `COPY`?

Cloning won't easily allow us to test out different branches.

Reply:
I think yes. (Though it's set to the wrong repo)
And if you're downloading this Dockerfile, you're not running the dev image, you're going to be running the prod image.

Reply:
Thinking more on this. Maybe not. Going to just use COPY and we'll see how clunky, if at all, it ends up being.

Reply:
If we do use COPY, we should copy package* files in a separate command, like [described here](https://nodejs.org/de/docs/guides/nodejs-docker-webapp/#creating-a-dockerfile).  Speeds up subsequent builds (assuming build layers are cached and dependencies haven't changed).

Also probably want to add a `.dockerignore` for `node_modules`.

Reply:
@codetheweb Broke it up even than that.

Reply:
Breaking up `COPY` commands doesn't really do much if there's nothing between them 😛 

I was thinking more like 
```dockerfile
WORKDIR /app/webapp/

COPY webapp/package*.json .

RUN npm install --only=prod

COPY webapp/* .

# Build up the frontend
ARG REACT_APP_ENDPOINT
ENV REACT_APP_ENDPOINT=$REACT_APP_ENDPOINT

RUN npm run build
```

Reply:
There are things between them?
Doing that your way would include things such as the README.md file that's under webapp/

Which then gets discarded later by only grabbing `/app/static/` from builder. Why even use the glob if all it does is result in discarded files?

Reply:
Yeah, you can change that last COPY if you want.

What I was trying to get at is that with cached build layers, a change in any of the COPYed files triggers all the commands below it to be re-run, instead of pulling from cache.  This means that if you copy the source code at the same time you copy `package.json`, a change in the source code will trigger `npm install` again (even though dependencies haven't changed).

http://bitjudo.com/blog/2014/03/13/building-efficient-dockerfiles-node-dot-js/

Reply:
Took me a bit of playing to understand what you were getting at. Think I finally got it though.

Reply:
Looks much better 👍.

The JSX comment just needs to be updated.
And I think line 9 can go before 7.

Reply:
I would disagree since the flow is:
1. Copy from source
2. CD into source
3. NPM install

Reply:
🤷 alright, doesn't really matter.